src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -0,0 +1,103 @@
+package seedu.duke.manager;
+
+import seedu.duke.command.BadCommandException;
+import seedu.duke.command.Command;
+import seedu.duke.command.CommandParser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Manager is the superclass for all WellNUS++ event drivers <br>
+ * <br>
+ * Each manager is in charge of 'managing' exactly one feature. <br>
+ * For example, <i>hb</i> and <i>reflect</i>. <br>
+ * <br>
+ * The manager should run an event driver (infinite loop) and is in charge
+ * of a Feature's input, output, 'business' logic and graceful termination
+ */
+public abstract class Manager {
+
+    private final CommandParser commandParser;
+    private ArrayList<Command> argumentList;
```

Comment:
Hmm argumentList is never initialised, this may lead to crashes since there's a getter method for it that's non-abstract.